Source/WebKit/UIProcess/LegacyGlobalSettings.h

```diff
@@ -52,6 +52,9 @@ class LegacyGlobalSettings {
     const HashSet<String>& schemesToRegisterAsNoAccess() { return m_schemesToRegisterAsNoAccess; }
     void registerURLSchemeAsNoAccess(const String& scheme) { m_schemesToRegisterAsNoAccess.add(scheme); }
 
+    const HashSet<String>& hostnamesToRegisterAsLocal() { return m_hostnamesToRegisterAsLocal; }
```

Comment:
This function should be marked as const. But also, it is suspicious to be adding new code to a class that is marked as "Legacy", these should probably be held somewhere else, maybe on the WebsiteDataStore.

Reply:
I looked into putting it into WebsiteDataStore but it seemed like an awkward fit.

This is used in `NetworkProcess::platformInitializeNetworkProcess()` which has access to `NetworkProcessCreationParameters` which is basically only filled with information from `LegacyGlobalSettings` and I didn't see clear access to any WebsiteDataStore.

Reply:
Maybe you can use WebsiteDataStoreCreationParameters instead? Maybe it won't work... but maybe it would.

WebsiteDataStore might be better named IAmTheNetworkProcessGodClassPutStuffInMe.

Even the network process itself is not named very well anymore. It's become the place to stuff anything that we don't want in the UI process or the web process.

Reply:
> Maybe you can use WebsiteDataStoreCreationParameters instead? Maybe it won't work... but maybe it would.

But don't consider that too closely, because it doesn't exist, and WebsiteDataStore is in the UI process. Ignore me....

Reply:
cc @achristensen07 who likely knows the best place to put this.

If Alex doesn't come back to you, just keep in LegacyGlobalSettings for now. I know WebsiteDataStore makes a lot more sense for Cocoa ports than glib ones because Cocoa uses a singleton network process (kept alive by the data store) while glib ports are still using the hold way.

Reply:
We used to have a NetworkProcess per WebProcessPool, owned by the WebProcessPool.

Now Cocoa port have a singleton NetworkProcessProxy, shared by all ProcessPools and kept alive by the WebsiteDataStore.

Reply:
Cocoa ports use:
```
    static Ref<NetworkProcessProxy> ensureDefaultNetworkProcess();
    static WeakPtr<NetworkProcessProxy>& defaultNetworkProcess();
```

Weak singleton. So we get at most one network process but this network process would go away if the last data store goes away.

Reply:
> Now Cocoa ports have a singleton NetworkProcessProxy, shared by all ProcessPools and kept alive via the WebsiteDataStores.

Ah... that wouldn't work for WPE/GTK because WebKitWebContext is a library context object: create a new WebKitWebContext and you should have an entirely separate instance of WebKit.

Reply:
Well... maybe it would work. Would just be... odd.

Reply:
CC @carlosgcampos who might be interested in this discussion